### PR TITLE
MOB-1786 (continued) Implemented date-based expiration date

### PR DESCRIPTION
**Pull request by d4r1091**
[MOB-1786](https://ecosia.atlassian.net/browse/MOB-1786)

## Context

I've got this proposal for you guys.

At the moment, we cache Derived Data at the branch level.

What does it mean 🤔 ?
It means that for every brand new PR, the build will start clean, without any Derived Data cache restored.
Everything else we cached is pulled accordingly (as usual) instead.
That's OK, we may think.
It already adds a level above our caching approach as much as possible.
But we can go beyond that ☝️.

## Approach

Looking at our process, I thought we would mutate the Derived Data cache from the branch-level to the `main` branch based.

The already thought 💭  edge case may be in case a feature (bookmarks, for instance) being pretty big, and the Derived Data cache may play a counter effect to caching, providing an old codebase reference to the feature we are building in CI.

To overcome that edge case, I thought this particular cache could be "date based."

The cache key will contain a date, among other values.
We'll utilize that, check against a target date (i.e., five days), and restore in case.
Otherwise, a new cache will be created.

The other cache won't be utilized; by Github rules, it will be automatically removed after seven days of stale.
This way, we'll maximize the caching process throughout all the cacheable components.

## Other

- Added a single commit we'll need to revert. It is utilized to trigger workflow checks.
- This PR also fixes the JUnit outputs ✅ 
- This PR also addresses a removal/skipping of a couple of UnitTests that were wrongly added to the TestPlans
- It also skip a single unit tests to review as it needs 20 seconds of expectation.

### CODE REVIEW

File: Client/Ecosia/MMP/MMP.swift [hunk 1]
```diff
@@ -6,6 +6,8 @@ import Foundation
 import Core
 import Shared
 
+// Test workflow trigger
```
Comment (lucaschifino) [inline Client/Ecosia/MMP/MMP.swift:9]: I'm guessing all those comments are meant only for triggering and should be removed before merging, just adding this comment as a reminder.

**Review by lucaschifino — COMMENTED** [review]
Looks promising 👏 🙏 

Added couple points on things I'm not 100% sure

File: Client/Ecosia/MMP/MMP.swift (hunk 1 above)
Comment (d4r1091) [reply Client/Ecosia/MMP/MMP.swift:9]: Correct 😉 - added in ## Other section

File: Tests/ClientTestPlan.xctestplan [hunk 2]
```diff
@@ -14,7 +14,10 @@
   "testTargets" : [
     {
       "skippedTests" : [
+        "DefaultSearchPrefsTests",
```
[the remark this answers is not in the document]
Comment (d4r1091) [reply Tests/ClientTestPlan.xctestplan:17]: The initial tests didn't add `DefaultSearchPrefsTests` and `HistoryDeletionUtilityTests` before moving onto `xctestplan`.
The `DefaultSearchPrefsTests` takes count on settings that aren't available at all as `ecosia` being the only search engine.
`HistoryDeletionUtilityTests` was added by mistake in the `xctestplan`.
I spent time trying to understand how to "fix" its failing tests, but after dedicating some time, I've decided to skip them simply. We'll bring them back later or perhaps after an upgrade.

**Review by lucaschifino — APPROVED** [review]
Green light from me, nice job 🟢 🚀

**Pull request merged**